security/entity_provider.rst

```diff
@@ -5,226 +5,25 @@
-            // $this->salt = md5(uniqid('', true));
-        }
-
-        public function getUsername()
-        {
-            return $this->username;
-        }
-
-        public function getSalt()
-        {
-            // you *may* need a real salt depending on your encoder
-            // see section on salt below
-            return null;
-        }
-
-        public function getPassword()
-        {
-            return $this->password;
-        }
-
-        public function getRoles()
-        {
-            return array('ROLE_USER');
-        }
-
-        public function eraseCredentials()
-        {
-        }
-
-        /** @see \Serializable::serialize() */
-        public function serialize()
-        {
-            return serialize(array(
-                $this->id,
-                $this->username,
-                $this->password,
-                // see section on salt below
-                // $this->salt,
-            ));
-        }
-
-        /** @see \Serializable::unserialize() */
-        public function unserialize($serialized)
-        {
-            list (
-                $this->id,
-                $this->username,
-                $this->password,
-                // see section on salt below
-                // $this->salt
-            ) = unserialize($serialized, array('allowed_classes' => false));
-        }
-    }
-
-To make things shorter, some of the getter and setter methods aren't shown.
-But you can generate these manually or with your own IDE.
-
-.. caution::
-
-    In the example above, the User entity's table name is "app_users" because
-    "USER" is a SQL reserved word. If you wish to call your table name "user",
-    `it must be quoted with backticks`_ to avoid errors. The annotation should
-    look like ``@ORM\Table(name="`user`")``.
-
-Next, make sure to :ref:`create the database table <doctrine-creating-the-database-tables-schema>`:
-
-.. code-block:: terminal
-
-    $ php bin/console doctrine:migrations:diff
-    $ php bin/console doctrine:migrations:migrate
-
-What's this UserInterface?
-~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-So far, this is just a normal entity. But to use this class in the
-security system, it must implement
-:class:`Symfony\\Component\\Security\\Core\\User\\UserInterface`. This
-forces the class to have the five following methods:
-
-* :method:`Symfony\\Component\\Security\\Core\\User\\UserInterface::getRoles`
-* :method:`Symfony\\Component\\Security\\Core\\User\\UserInterface::getPassword`
-* :method:`Symfony\\Component\\Security\\Core\\User\\UserInterface::getSalt`
-* :method:`Symfony\\Component\\Security\\Core\\User\\UserInterface::getUsername`
-* :method:`Symfony\\Component\\Security\\Core\\User\\UserInterface::eraseCredentials`
-
-To learn more about each of these, see :class:`Symfony\\Component\\Security\\Core\\User\\UserInterface`.
-
-.. caution::
-
-    The ``eraseCredentials()`` method is only meant to clean up possibly stored
```

Comment:
I think we need to keep this caution somewhere

Reply:
Hmm. We don't specifically talk at all about this method now. And, that's on purpose - it's not something most users need to know/care about.

What if we added the note in the generated code in MakerBundle? Also, I also think it would be a good idea to *show* the generated `User` class here... but I would want it in some "collapsed" format so that you could see it if you wanted to, but it wouldn't be visible (and taking up a lot of space) by default. That feature doesn't exist on the docs yet. But when it does, we would show the generated code, and it would include that warning (if we did my idea).